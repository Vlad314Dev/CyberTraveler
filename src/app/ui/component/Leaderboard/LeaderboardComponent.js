import './LeaderboardStyle';

import React, { PureComponent } from 'react';
import LeaderboardList from 'UIComponent/LeaderboardList';

class LeaderboardComponent extends PureComponent
{
    render()
    {
        return (
            <div block="Leaderboard">
                <div block="Leaderboard" elem="Title">TOP 10</div>
                <div block="Leaderboard" elem="UserRank">Authorize to see your current rank</div>
                <LeaderboardList/>
            </div>
        )
    }
}

export default LeaderboardComponent;
