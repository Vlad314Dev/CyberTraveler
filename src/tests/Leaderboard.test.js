/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import LeaderboardComponent from '../app/ui/component/Leaderboard/LeaderboardComponent';

describe('Leaderboard', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <LeaderboardComponent/>
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <LeaderboardComponent/>
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
