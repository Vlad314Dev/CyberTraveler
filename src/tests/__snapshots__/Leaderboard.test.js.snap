// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard Should render layout 1`] = `
Array [
  <div
    className="Leaderboard"
  >
    <div
      className="Leaderboard__Title"
    >
      TOP 10
    </div>
    <Memo(Connect(LeaderboardListContainer)) />
  </div>,
]
`;
